modules/OFStateManager/module/src/handlers.c

```diff
@@ -312,6 +312,18 @@ ind_core_queue_stats_request_handler(of_object_t *_obj, indigo_cxn_id_t cxn_id)
 
         of_queue_stats_entry_delete(queue_stats);
         indigo_cxn_send_controller_message(cxn_id, reply);
+    } else if (indigo_port_queue_stats_get_handler) {
+        rv = indigo_port_queue_stats_get_handler(obj, cxn_id);
+        if (rv == INDIGO_ERROR_NONE) {
```

Comment:
it looks like we're calling indigo_port_queue_stats_get_handler if that function is defined.
but if that function returns INDIGO_ERROR_NONE, then we're going to send an error reply?
maybe i read this code too quickly and missed something obvious.

Reply:
Originally, I planned not to pass the cxn_id to the handler() so that the error status is handled in indigo. If handler() function is ok to return the error message, then indigo can do nothing when error is returned.

Reply:
I will let the handler() to return the error as suggested

Reply:
i think we should provide some more detail for the comment that precedes indigo_port_queue_stats_get_handler in port_manager.h.  if the handler is supposed to deal with any error, then the prototype of the handler should return void, and the comment should say "handler should send reply or error".

alternately, you could keep the handler prototype as it is, and your code in handler.c can send an error with appropriate type and code if the rv != INDIGO_ERROR_NONE.  so if the driver wants to send an error, it can do so inside the handler, and the handler can return INDIGO_ERROR_NONE so that no error is sent by OFStateManager.

so just pick one or the other, but from your comment above, it sounds like you're going with the first option;  can you confirm?